src/editor/mod.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::Path;
+use std::fs::File;
+use std::io::Read;
+
+pub static JS: &'static [u8] = include_bytes!("editor.js");
+pub static ACE_JS: &'static [u8] = include_bytes!("ace.js");
+pub static MODE_RUST_JS: &'static [u8] = include_bytes!("mode-rust.js");
+pub static THEME_DAWN_JS: &'static [u8] = include_bytes!("theme-dawn.js");
+pub static THEME_TOMORROW_NIGHT_JS: &'static [u8] = include_bytes!("theme-tomorrow_night.js");
+
+/// Integration of a JavaScript editor for playpens.
+/// Uses the Ace editor: https://ace.c9.io/.
+/// The Ace editor itself, the mode, and the theme files are the
+/// generated minified no conflict versions.
+///
+/// The `Editor` struct should be used instead of the static variables because
+/// the `new()` method
+/// will look if the user has an editor directory in his source folder and use
+/// the users editor instead
```

Comment:
I think we misunderstood each other. I am not talking about the code here, I am talking about fetching the resource files. If I understand this PR correctly, you would allow the user to define a new directory in their root folder called `editor` containing the resource files the user wants to override. I'm merely proposing to **not** use a new directory for that but use the existing theme directory convention.

Is there something I am misunderstanding?